BlueprintUICommonControls/Sources/Box.swift

```diff
@@ -334,4 +334,39 @@ fileprivate final class BoxView: UIView {
         }
     }
 
+    /// This method is overridden to provide a `ShadowPathAction` for the `shadowPath` event;
+    /// this ensure the shadow implicitly animates alongside changes in the views size.
+    override func action(for layer: CALayer, forKey event: String) -> CAAction? {
+        let keyPath = #keyPath(CALayer.shadowPath)
+
+        guard event == keyPath,
+              let currentPath = layer.shadowPath,
+              let sizeAnimation = layer.animation(forKey: "bounds.size") as? CABasicAnimation
+        else {
+            return super.action(for: layer, forKey: event)
+        }
+
+        let animation = sizeAnimation.copy() as! CABasicAnimation
+        animation.keyPath = keyPath
+        animation.fromValue = currentPath
+
+        return ShadowPathAction(pendingAnimation: animation)
```

Comment:
Can you not return `animation` itself here? I’ve done that in the past with success

Reply:
How do you set the `toValue`? Do you not need to since the actual value is set?

Reply:
Huh, it certainly seems to work!

Reply:
Yeah, tbh this is one of those mysterious/under-documented parts of CoreAnimation. But `toValue` seems to happen magically in there somewhere.